internal/provider/vaultsecrets/rotating_secret_confluent.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vaultsecrets
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/preview/2023-11-28/client/secret_service"
+	secretmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/preview/2023-11-28/models"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+var _ rotatingSecret = &confluentRotatingSecret{}
+
+type confluentRotatingSecret struct{}
+
+func (s *confluentRotatingSecret) read(ctx context.Context, client secret_service.ClientService, secret *RotatingSecret) (any, error) {
+	response, err := client.GetConfluentCloudAPIKeyRotatingSecretConfig(
+		secret_service.NewGetConfluentCloudAPIKeyRotatingSecretConfigParamsWithContext(ctx).
+			WithOrganizationID(secret.OrganizationID.ValueString()).
+			WithProjectID(secret.ProjectID.ValueString()).
+			WithAppName(secret.AppName.ValueString()).
+			WithName(secret.Name.ValueString()), nil)
+	if err != nil && !clients.IsResponseCodeNotFound(err) {
+		return nil, err
+	}
+	if response == nil || response.Payload == nil {
+		return nil, nil
+	}
+	return response.Payload.Config, nil
+}
+
+func (s *confluentRotatingSecret) create(ctx context.Context, client secret_service.ClientService, secret *RotatingSecret) (any, error) {
+	if secret.ConfluentServiceAccount == nil {
+		return nil, fmt.Errorf("missing required field 'confluent_service_account'")
+	}
+
+	response, err := client.CreateConfluentCloudAPIKeyRotatingSecret(
+		secret_service.NewCreateConfluentCloudAPIKeyRotatingSecretParamsWithContext(ctx).
+			WithOrganizationID(secret.OrganizationID.ValueString()).
+			WithProjectID(secret.ProjectID.ValueString()).
+			WithAppName(secret.AppName.ValueString()).
+			WithBody(&secretmodels.SecretServiceCreateConfluentCloudAPIKeyRotatingSecretBody{
+				IntegrationName:    secret.IntegrationName.ValueString(),
+				RotationPolicyName: secret.RotationPolicyName.ValueString(),
+				ConfluentCloudAPIKeyParams: &secretmodels.Secrets20231128ConfluentCloudAPIKeyParams{
+					ServiceAccountID: secret.ConfluentServiceAccount.ServiceAccountID.ValueString(),
+				},
+				Name: secret.Name.ValueString(),
+			}),
+		nil)
+	if err != nil {
+		return nil, err
+	}
+	if response == nil || response.Payload == nil {
+		return nil, nil
+	}
+	return response.Payload.Config, nil
+}
```

Comment:
I got this error when trying to force a recreate by changing the Service Account ID:
```
Terraform will perform the following actions:

  # hcp_vault_secrets_rotating_secret.example_confluent must be replaced
-/+ resource "hcp_vault_secrets_rotating_secret" "example_confluent" {
      ~ confluent_service_account = {
          ~ service_account_id = "sa-9mo167" -> "sa-5r3k2z" # forces replacement
        }
        name                      = "my_confluent_1"
      ~ organization_id           = "1b7bb3fd-7189-4bce-9016-7b2f53515b13" -> (known after apply)
      ~ project_id                = "db3e1be5-60cb-450d-a758-5663d3a81771" -> (known after apply)
        # (4 unchanged attributes hidden)
    }

Plan: 1 to add, 0 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

hcp_vault_secrets_rotating_secret.example_confluent: Destroying... [name=my_confluent_1]
hcp_vault_secrets_rotating_secret.example_confluent: Destruction complete after 0s
hcp_vault_secrets_rotating_secret.example_confluent: Creating...
╷
│ Error: Error creating Vault Secrets resource
│ 
│   with hcp_vault_secrets_rotating_secret.example_confluent,
│   on rotating_secrets.tf line 45, in resource "hcp_vault_secrets_rotating_secret" "example_confluent":
│   45: resource "hcp_vault_secrets_rotating_secret" "example_confluent" {
│ 
│ [POST /secrets/2023-11-28/organizations/{organization_id}/projects/{project_id}/apps/{app_name}/rotating/confluent/secret][409] CreateConfluentCloudApiKeyRotatingSecret default
│ &{Code:6 Details:[] Message:a secret with name "my_confluent_1" already exists}
```

A subsequent `terraform apply` passed so it's a timing issue where the create request was fire before the delete operation was completed. Is there any async component on the HVS side when deleting a rotating secret?

I don't think it is exclusively related to this PR but if someone can reproduce it'd be worth logging a bug to investigate how to make recreates graceful.

Reply:
Edit: nvm it's because I used hardcoded name reference instead of proper Terraform attribute references.